lms/services/lti_grading/_v13.py

```diff
@@ -56,13 +59,26 @@ def record_result(self, grading_id, score=None, pre_record_hook=None):
         if pre_record_hook:
             payload = pre_record_hook(score=score, request_body=payload)
 
-        return self._ltia_service.request(
-            "POST",
-            self._service_url(self.line_item_url, "/scores"),
-            scopes=self.LTIA_SCOPES,
-            json=payload,
-            headers={"Content-Type": "application/vnd.ims.lis.v1.score+json"},
-        )
+        try:
+            return self._ltia_service.request(
+                "POST",
+                self._service_url(self.line_item_url, "/scores"),
+                scopes=self.LTIA_SCOPES,
+                json=payload,
+                headers={"Content-Type": "application/vnd.ims.lis.v1.score+json"},
+            ).json()
+
+        except ExternalRequestError as err:
+            if (
+                err.status_code == 422
+                and "maximum number of allowed attempts has been reached"
+                in err.response.text
+            ):
+                LOG.error("record_result: maximum number of allowed attempts")
+                # We silently shallow this type of error
+                return {}
```

Comment:
If we eventually want to show some modal to the student we can just make a change here.

I reckon an error modal might not be the right choice thought. This happens while creating the first annotation in a "session" so it's no clear for the student that they are in fact creating a submission.

They do get an error modal the moment they make that annotation but the annotation was made successful so I reckon this change is less confusing.

See https://hypothes-is.slack.com/archives/C4K6M7P5E/p1675419187872529 for further discussion.